src/VisualStudio/IntegrationTest/IntegrationTests/VisualBasic/BasicChangeSignatureDialog.cs

```diff
@@ -36,31 +40,117 @@ End Sub
         public void VerifyRefactoringCancelled()
         {
+
             SetUpEditor(@"
 Class C
     Sub Method$$(a As Integer, b As String)
     End Sub
 End Class");
 
-            OpenDialog();
+            InvokeChangeSignature();
             VerifyChangeSignatureDialog(isOpen: true);
-            ChangeSignatureDialog_ClickCancel();
+            ClickCancel();
             VerifyChangeSignatureDialog(isOpen: false);
+
+            VerifyTextContains(@"
+Class C
+    Sub Method(a As Integer, b As String)
+    End Sub
+End Class");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ChangeSignature)]
+        public void VerifyReorderParameters()
+        {
+            SetUpEditor(@"
+Class C
+    Sub Method$$(a As Integer, b As String)
+    End Sub
+End Class");
+
+            InvokeChangeSignature();
+            VerifyChangeSignatureDialog(isOpen: true);
+            SelectParameter("Integer a");
+            ClickDownButton();
+            ClickOK();
+            VerifyChangeSignatureDialog(isOpen: false);
+
+            VerifyTextContains(@"
+Class C
+    Sub Method(b As String, a As Integer)
+    End Sub
+End Class");
         }
 
-        private void OpenDialog()
+        private void InvokeChangeSignature()
         {
             Editor.SendKeys(Ctrl(VirtualKey.R), Ctrl(VirtualKey.V));
         }
 
         private void VerifyChangeSignatureDialog(bool isOpen)
         {
-            VerifyDialog("ChangeSignatureDialog", isOpen);
+            VerifyDialog(ChangeSignatureDialogAutomationId, isOpen);
+        }
+
+        private void ClickOK()
+        {
+            PressDialogButton(ChangeSignatureDialogAutomationId, "OKButton");
         }
 
-        private void ChangeSignatureDialog_ClickCancel()
+        private void ClickCancel()
         {
-            PressDialogButton("ChangeSignatureDialog", "CancelButton");
+            PressDialogButton(ChangeSignatureDialogAutomationId, "CancelButton");
+        }
+
+        private void ClickDownButton()
+        {
+            PressDialogButton(ChangeSignatureDialogAutomationId, "DownButton");
+        }
+
+        private void ClickUpButton()
+        {
+            PressDialogButton(ChangeSignatureDialogAutomationId, "UpButton");
+        }
+
+        private void SelectParameter(string parameterName)
+        {
+            var dialogAutomationElement = GetDialog(ChangeSignatureDialogAutomationId);
+
+            Condition propertyCondition = new PropertyCondition(AutomationElement.AutomationIdProperty, "MemberSelectionList");
+            var grid = dialogAutomationElement.FindFirst(TreeScope.Descendants, propertyCondition);
+
+            var gridPattern = grid.GetCurrentPattern(GridPattern.Pattern) as GridPattern;
+            var rowCount = (int)grid.GetCurrentPropertyValue(GridPattern.RowCountProperty);
+            var columnToSelect = 2;
+            int i = 0;
+            for (; i < rowCount; i++)
```

Comment:
Why not just put the int I = 0; in the initializer section of the for loop? It looks like the only use outside is the `if (I == rowCount)` which is always true?